FinSearchUnified/Tests/Subscriber/FrontendTest.php

```diff
@@ -234,6 +296,47 @@ public function testBeforeSearchIndexAction($vendor)
         $frontend->beforeSearchIndexAction($args);
     }
 
+    /**
+     * @dataProvider legacyUrlProvider
+     *
+     * @param array $params
+     * @param string $expectedUrl
+     */
+
+    public function testLegacyUrls(array $params, $expectedUrl)
+    {
+        // Create Request object to be passed in the mocked Subject
+        $request = new RequestHttp();
+        $request->setControllerName($params['controller'])
+            ->setActionName($params['action'])
+            ->setQuery($params)
+            ->setBaseUrl(rtrim(Shopware()->Shop()->getHost(), ' / '));
+
+        // Create mocked Subject to be passed in mocked args
+        $subject = $this->getMockBuilder(Enlight_Controller_Action::class)
+            ->setMethods(['Request', 'redirect'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $subject->method('Request')
+            ->willReturn($request);
+
+        if ($expectedUrl === null) {
+            $subject->expects($this->never())->method('redirect');
+        } else {
+            $subject->method('redirect')->with($expectedUrl, ['code' => 301]);
+        }
+
+        // Create mocked args for getting Subject and Request
+        $args = $this->getMockBuilder(Enlight_Event_EventArgs::class)
```

Comment:
* Generally speaking: You should prefer the actual implementation over mocks. Sometimes it's not that easy to get an actual instance (e.g. because of dependencies), in which case we have to settle for a mock.
* In this case: `Enlight_Controller_Action` and `Enlight_Event_EventArgs` can be used directly and don't have to be mocked.
* Check `FrontendTest::testSessionValuesAfterListingCount`.